pkg/common/resource.go

```diff
@@ -72,9 +72,32 @@ func GetPodResource(pod *v1.Pod) (resource *si.Resource) {
 		containerResource := getResource(resourceList)
 		podResource = Add(podResource, containerResource)
 	}
+
+	// vcore, mem compare between initcontainer and containers and replace
+	// For each resource, podResourceRequest = max(sum(Containers requirement), InitContainers requirement)
+	if pod.Spec.InitContainers != nil {
+		isNeedMoreResourceAndSet(pod, podResource)
+	}
+
 	return podResource
 }
 
+func isNeedMoreResourceAndSet(pod *v1.Pod, containersResources *si.Resource) {
+	for _, c := range pod.Spec.InitContainers {
+		resourceList := c.Resources.Requests
+		ICResource := getResource(resourceList)
+		for resourceName, ICRequest := range ICResource.Resources {
+			containersRequests, exist := containersResources.Resources[resourceName]
```

Comment:
Does this mean we skip a request that is not defined on the container resources? That does not look right. 
cont = 1GB, 1cpu
init cont = 2GB, 1GPU
the result should be 2GB, 1cpu, 1GPU a merge of all requests.

Reply:
I will add test for this situation in next commit.
In my thought, it will be like following
loop1 men:2GB > 1GB replace.
loop2 GPU is not found in cont request, additional request add.

Reply:
Ok

Reply:
Thanks @wilfred-s.
Additional resource from init cont  also add into request in next commit.